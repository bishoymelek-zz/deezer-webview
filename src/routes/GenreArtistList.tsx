import React from 'react';
import ArtistCard from 'components/ArtistCard';
import { genresActionCreators } from 'state-config/genreStore';
import { connect } from 'react-redux';
import { Row, Col } from 'antd';
import Spinner from 'components/Spinner';
import { withRouter } from 'react-router';

type GenreArtistListPropType = {
  genreArtistsList: Array<deezerArtistObjectPropType>;
  genreArtistsListApiLoading: boolean;
};
class GenreArtistListComponent extends React.Component<
  componentGenericPropType &
    connectedComponentPropType &
    GenreArtistListPropType,
  {}
> {
  componentDidMount(): void {
    const { dispatch } = this.props;
    dispatch(
      genresActionCreators.getGenreArtistsList(
        this.props?.match?.params?.genreId
      )
    );
  }

  render(): JSX.Element {
    const { genreArtistsList = [], genreArtistsListApiLoading } = this.props;
    return (
      <>
        <Spinner spinning={genreArtistsListApiLoading} />
        <Row gutter={[24, 24]} className="p-2">
          {genreArtistsList.length
            ? genreArtistsList.map((genreItem) => {
                return (
                  <Col xs={24} sm={12} md={8} lg={6} key={genreItem.id}>
                    <ArtistCard
                      id={genreItem.id}
                      trackListUrl={genreItem.tracklist}
                      name={genreItem.name}
                      picture={genreItem?.picture_big}
                    />
                  </Col>
                );
              })
            : null}
        </Row>
      </>
    );
  }
}

const mapStateToProps = (state: any): GenreArtistListPropType => ({
  genreArtistsList: state.api_genres?.genreArtistsList,
  genreArtistsListApiLoading: state.api_genres?.status === 'loading',
});

const GenreArtistList = connect(mapStateToProps)(
  withRouter(GenreArtistListComponent)
);

export default GenreArtistList;
