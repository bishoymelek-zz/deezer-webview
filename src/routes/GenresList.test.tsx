import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import GenreArtistList from './GenreArtistList';

const mockStore = configureStore([]);

describe('Route/GenreArtistList', () => {
  let store;
  let component;
  const GenreArtistListMockData = {
    name: 'Hi',
    id: '1',
    picture: 'pic.png',
    trackListUrl: 'https://placeholder.com',
  };
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <GenreArtistList
            name={GenreArtistListMockData.name}
            id={GenreArtistListMockData.id}
            picture={GenreArtistListMockData.picture}
            trackListUrl={GenreArtistListMockData.trackListUrl}
          />
        </BrowserRouter>
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
