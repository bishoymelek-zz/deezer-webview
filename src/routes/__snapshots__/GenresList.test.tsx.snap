// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route/GenreArtistList should render with given state from Redux store 1`] = `
Array [
  <div
    className="ant-page-header site-page-header ant-page-header-ghost"
  >
    <div
      className="ant-page-header-heading"
    >
      <div
        className="ant-page-header-heading-left"
      >
        <span
          className="ant-page-header-heading-title"
          title="Genre Top Artists"
        >
          Genre Top Artists
        </span>
        <span
          className="ant-page-header-heading-sub-title"
          title="Choose your favorite artist"
        >
          Choose your favorite artist
        </span>
      </div>
    </div>
  </div>,
  <div
    className="ant-row p-2"
    style={
      Object {
        "marginBottom": 12,
        "marginLeft": -12,
        "marginRight": -12,
        "marginTop": -12,
      }
    }
  />,
]
`;
